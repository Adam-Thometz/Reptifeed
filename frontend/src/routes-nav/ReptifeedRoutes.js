import React from "react";
import { Routes, Route } from 'react-router-dom'

import Home from "../home/Home";
import Login from "../auth/Login";
import Register from "../auth/Register";
import Profile from "../user/Profile";
import EditUserForm from "../user/EditUserForm";
import Reptiles from "../reptile/Reptiles";
import NewReptileForm from "../reptile/NewReptileForm";

import Foods from "../food/Foods";
import Pantry from "../pantry/Pantry";
import NotFound from "../common/NotFound";
import Reptile from "../reptile/Reptile";
import EditReptileForm from "../reptile/EditReptileForm";
import RequireAuth from "./RequireAuth";

const ReptifeedRoutes = ({ login, register }) => {
  return (
    <div className="Routes">
      <Routes>
        {/* Home page */}
        <Route index element={<Home />} />

        {/***************/}
        {/* AUTH ROUTES */}
        {/***************/}

        {/* Login page */}
        <Route path="/login" element={<Login login={login} />} />
        {/* Registration page */}
        <Route path="/register" element={<Register register={register} />} />

        {/***************/}
        {/* USER ROUTES */}
        {/***************/}

        {/* PRIVATE ROUTE: Profile page */}
        <Route path="/users/:id" element={<RequireAuth><Profile /></RequireAuth>} />
        {/* PRIVATE ROUTE: Edit profile */}
        <Route path="/users/:id/edit" element={<RequireAuth><EditUserForm /></RequireAuth>} />

        {/******************/}
        {/* REPTILE ROUTES */}
        {/******************/}

        {/* PRIVATE ROUTE: View reptiles */}
        <Route path="/users/:id/reptiles" element={<RequireAuth><Reptiles /></RequireAuth>} />
        {/* PRIVATE ROUTE: Add a new reptile */}
        <Route path="/users/:id/reptiles/add" element={<RequireAuth><NewReptileForm /></RequireAuth>} />
        {/* PRIVATE ROUTE: View a specific reptile */}
        <Route path="/users/:id/reptiles/:reptileId" element={<RequireAuth><Reptile /></RequireAuth>} />
        {/* PRIVATE ROUTE: Edit a specific reptile */}
        <Route path="/users/:id/reptiles/:reptileId/edit" element={<RequireAuth><EditReptileForm /></RequireAuth>} />

        {/**********************/}
        {/* FOOD/PANTRY ROUTES */}
        {/**********************/}
        
        {/* PRIVATE ROUTE: View all available foods for reptiles */}
        <Route path="/foods" element={<Foods />} />
        {/* PRIVATE ROUTE: View pantry */}
        <Route path="/users/:id/pantry" element={<RequireAuth><Pantry /></RequireAuth>} />

        {/* GENERAL 404 HANDLER */}
        <Route path="*" element={<NotFound />} />
      </Routes>
    </div>
  );
};

export default ReptifeedRoutes;