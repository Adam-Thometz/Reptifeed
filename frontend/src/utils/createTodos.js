
/** createTodos(reptiles, pantry)
 * 
 * Takes a user's reptiles and pantry and creates a list of things they need to do to unlock the app's full features
 * 
 * Returns an two arrays of Link components, one of important things and one of nice-to-haves
 */

function createTodos(reptiles, pantry) {
  /** Essential checks:
   * - reptileCheck: Checks if a user has reptiles
   * - vegetableOftenCheck: Checks if a user has a vegetable staple in their pantry
   * - proteinOftenCheck: Checks if a user has a protein staple in their pantry
   * - treatModeratelyCheck: Checks if a user has a treat staple in their pantry
   * - supplementCheck: Checks if a user has at least two different supplements
   */
  const essentialTodos = [];

  const reptileCheck = !!reptiles.length;
  const vegetableOftenCheck = pantry.some(food => food.type === 'vegetable' && food.frequency === 'often');
  const proteinOftenCheck = pantry.some(food => food.type === 'protein' && food.frequency === 'often');
  const treatModeratelyCheck = pantry.some(food => food.isTreat && food.frequency === 'moderately');
  const supplementCheck = (pantry.filter(food => food.type === 'supplement')).length >= 2;

  if (!reptileCheck) essentialTodos.push('Add a reptile');
  if (!vegetableOftenCheck) essentialTodos.push('Add vegetables eaten often');
  if (!proteinOftenCheck) essentialTodos.push('Add protein eaten often');
  if (!treatModeratelyCheck) essentialTodos.push('Add a treat eaten moderately');
  if (!supplementCheck) essentialTodos.push('Add at least two supplements');

  /** Nice-to-have checks:
   * - vegetableModeratelyCheck: Checks if a user has a vegetable that's eaten moderately in their pantry
   * - vegetableOccasionallyCheck: Checks if a user has a vegetable that's eaten occasionally in their pantry
   * - proteinModeratelyCheck: Checks if a user has a protein that's eaten moderately in their pantry
   * - proteinOccasionallyCheck: Checks if a user has a protein that's eaten occasionally in their pantry
   * - treatOccasionallyCheck: Checks if a user has a treat that's eaten rarely in their pantry
   */
  const niceToHaveTodos = [];

  const vegetableModeratelyCheck = pantry.some(food => food.type === 'vegetable' && food.frequency === 'moderately');
  const vegetableOccasionallyCheck = pantry.some(food => food.type === 'vegetable' && food.frequency === 'occasionally' && !food.isTreat);
  const proteinModeratelyCheck = pantry.some(food => food.type === 'protein' && food.frequency === 'moderately');
  const proteinOccasionallyCheck = pantry.some(food => food.type === 'protein' && food.frequency === 'occasionally' && !food.isTreat);
  const treatOccasionallyCheck = pantry.some(food => food.isTreat && food.frequency === 'occasionally');

  if (!vegetableModeratelyCheck) niceToHaveTodos.push('Add a vegetable eaten moderately');
  if (!vegetableOccasionallyCheck) niceToHaveTodos.push('Add a vegetable eaten occasionally');
  if (!proteinModeratelyCheck) niceToHaveTodos.push('Add a protein eaten moderately');
  if (!proteinOccasionallyCheck) niceToHaveTodos.push('Add a protein eaten occasionally');
  if (!treatOccasionallyCheck) niceToHaveTodos.push('Add a treat eaten occasionally');

  return {essentialTodos, niceToHaveTodos};
};

export default createTodos;