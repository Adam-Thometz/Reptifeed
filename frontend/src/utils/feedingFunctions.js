/** Feeding functions
 * 
 * Used to get foods from pantry that match frequencies and food type.
*/

function getFoodFromPantry(pantry, freq, type) {
  const availableFood = pantry.filter(food => (!food.isTreat && (food.frequency === freq && food.type === type)));
  if (!availableFood.length) return `Get some more ${type}`
  const selectedIdx = Math.floor(Math.random() * availableFood.length);

  return availableFood[selectedIdx];
};

function getTreatFromPantry(pantry, freq) {
  const availableTreats = pantry.filter(food => food.isTreat && food.frequency === freq);
  if (!availableTreats.length) return "You don't have any treats";
  const selectedIdx = Math.floor(Math.random() * availableTreats.length); 
  
  return availableTreats[selectedIdx];
};

export { getFoodFromPantry, getTreatFromPantry };