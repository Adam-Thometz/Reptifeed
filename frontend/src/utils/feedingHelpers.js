/** Feeding functions
 * 
 * Used to get foods from pantry that match frequencies and food type.
 * 
 * If no foods are available for the frequency, returns foods eaten most often
*/

function getFoodFromPantry(pantry, freq, type) {
  let availableFood = pantry.filter(food => (!food.isTreat && (food.frequency === freq && food.type === type)));
  if (!availableFood.length) {
    availableFood = pantry.filter(food => (!food.isTreat && (food.frequency === 'often' && food.type === type)));
  };
  const selectedIdx = Math.floor(Math.random() * availableFood.length);

  return availableFood[selectedIdx];
};

function getTreatFromPantry(pantry, freq) {
  let availableTreats = pantry.filter(food => food.isTreat && food.frequency === freq);
  if (!availableTreats.length) {
    availableTreats = pantry.filter(food => food.isTreat);
  };
  const selectedIdx = Math.floor(Math.random() * availableTreats.length); 
  
  return availableTreats[selectedIdx];
};

function getNextSupplement(pantry, lastSupplement) {
  const nextSupplement = lastSupplement === 'calcium' ?
    pantry.find(food => food.name === 'multivitamin') :
    pantry.find(food => food.name === 'calcium');
  
  return nextSupplement;
}

export { getFoodFromPantry, getTreatFromPantry, getNextSupplement };