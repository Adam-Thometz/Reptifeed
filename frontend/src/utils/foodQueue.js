// foodQueue: Feed reptile based on frequency
const foodQueue = ["often", "often", "moderately", "often", "often", "moderately", "often", "often", "moderately", "occasionally"];

// treatQueue: Give reptile treats based on frequency
const treatQueue = ["moderately", "moderately", "occasionally"];

export { foodQueue, treatQueue };