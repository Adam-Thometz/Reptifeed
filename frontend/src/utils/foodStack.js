// foodStack: Feed reptile based on frequency
const foodStack = ["occasionally", "moderately", "often", "often", "moderately", "often", "often", "moderately", "often", "often"];

// treatStack: Give reptile treats based on frequency
const treatStack = ["occasionally", "moderately", "moderately"];

export { foodStack, treatStack };