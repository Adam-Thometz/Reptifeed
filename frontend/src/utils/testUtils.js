import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: 'rickSanchez',
  password: 'ilikeMorty',
  email: 'rickman@qmail.com',
  isAdmin: false
};

const demoReptile = [
  {
    id: 1,
    name: 'two crows',
    species: 'crows',
    subspecies: 'two of them',
    birthday: '2015-10-29',
    imgUrl: '',
    ownerId: 1
  }
];

const demoPantry = [
  {
    name: 'plumbus',
    type: 'everything',
    frequency: 'always',
    image: '',
    isTreat: true,
    tips: "Everyone needs a plumbus!"
  }
];

const UserProvider = ({ children, currUser = demoUser, pantry = demoPantry, reptiles = demoReptile }) => (
  <UserContext.Provider value={{currUser, pantry, reptiles}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };