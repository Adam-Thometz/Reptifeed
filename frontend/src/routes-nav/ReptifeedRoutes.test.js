import React from "react";
import ReptifeedRoutes from "./ReptifeedRoutes";
import { render, UserProvider } from "../utils/testUtils";

test('it renders without crashing', () => {
  render(<ReptifeedRoutes />);
});

test('it matches snapshot', () => {
  const { asFragment } = render(<ReptifeedRoutes />);
  expect(asFragment()).toMatchSnapshot();
});

test('it matches snapshot of login page', () => {
  const { asFragment } = render(<ReptifeedRoutes />, { initialRoutes: ['/login'] });
  expect(asFragment()).toMatchSnapshot();
});

test('it matches snapshot of registration page', () => {
  const { asFragment } = render(<ReptifeedRoutes />, { initialRoutes: ['/register'] });
  expect(asFragment()).toMatchSnapshot();
});

test('it matches snapshot of profile page', () => {
  const { asFragment } = render(<UserProvider><ReptifeedRoutes /></UserProvider>, { initialRoutes: ['/users/1'] });
  expect(asFragment()).toMatchSnapshot();
});

test('it matches snapshot of edit user page', () => {
  const { asFragment } = render(<UserProvider><ReptifeedRoutes /></UserProvider>, { initialRoutes: ['/users/1/edit'] });
  expect(asFragment()).toMatchSnapshot();
});

test('it matches snapshot of 404 page', () => {
  const { asFragment } = render(<UserProvider><ReptifeedRoutes /></UserProvider>, { initialRoutes: ['/hello-lol'] });
  expect(asFragment()).toMatchSnapshot();
});