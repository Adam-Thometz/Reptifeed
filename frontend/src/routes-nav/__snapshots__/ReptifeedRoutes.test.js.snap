// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Home"
    >
      <div
        class="Home-background"
      />
      <div
        class="Home-content"
      >
        <h1>
          Welcome to Reptifeed!
        </h1>
        <p>
          Your personal assistant for all things reptile food!
        </p>
        <div
          class="Home-content-description"
        >
          <p>
            You are here because you want what's best for your pet.
          </p>
          <img
            alt="A bearded dragon perched on a log and looking into the camera with its mouth slightly open"
            src="looking.jpeg"
          />
        </div>
        <div
          class="Home-content-description"
        >
          <img
            alt="A tortoise about to eat a dandelion"
            src="tortoise.jpeg"
          />
          <p>
            You know that a diverse selection of food is key to good health.
          </p>
        </div>
        <div
          class="Home-content-description"
        >
          <p>
            Keep an inventory of your food with us and we'll help you decide what to feed your reptile.
          </p>
          <img
            alt="A bearded dragon about to eat from a bowl of greens and insects"
            src="bearded-dragon.jpeg"
          />
        </div>
        <div
          class="Home-content-description"
        >
          <img
            alt="A blue tongue skink eating dog food and an egg"
            src="blue-tongue-skink.png"
          />
          <p>
            We'll help you vary your reptile's diet so they can thrive.
          </p>
        </div>
        <div
          class="Home-cta"
        >
          <h4>
            Get started with us!
          </h4>
          <button>
            Register
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of 404 page 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="NotFound"
    >
      <h4>
        The page you're looking for wasn't found
      </h4>
      <a
        href="/"
      >
        Go back home
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of add new reptile 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="NewReptileForm"
    >
      <h1>
        Add a reptile to your account
      </h1>
      <form>
        <div
          class="NewReptileForm-form-group"
        >
          <label
            class="NewReptileForm-label"
            for="name"
          >
            Name:
          </label>
          <input
            class="NewReptileForm-input"
            id="name"
            name="name"
            type="text"
            value=""
          />
        </div>
        <div
          class="NewReptileForm-form-group"
        >
          <label
            class="NewReptileForm-label"
            for="species"
          >
            Species:
          </label>
          <select
            class="NewReptileForm-input"
            id="species"
            name="species"
          >
            <option
              value=""
            >
              -
            </option>
            <option
              value="blue tongue skink"
            >
              Blue Tongue Skink
            </option>
          </select>
        </div>
        <div
          class="NewReptileForm-form-group"
        >
          <label
            class="NewReptileForm-label"
            for="subspecies"
          >
            Subspecies:
          </label>
          <input
            class="NewReptileForm-input"
            id="subspecies"
            name="subspecies"
            type="text"
            value=""
          />
        </div>
        <div
          class="NewReptileForm-form-group"
        >
          <label
            class="NewReptileForm-label"
            for="birthday"
          >
            Birthday:
          </label>
          <input
            class="NewReptileForm-input"
            id="birthday"
            name="birthday"
            type="date"
            value=""
          />
        </div>
        <div
          class="NewReptileForm-form-group"
        >
          <label
            class="NewReptileForm-label"
            for="imgUrl"
          >
            Picture:
          </label>
          <input
            class="NewReptileForm-input"
            id="imgUrl"
            name="imgUrl"
            type="text"
            value=""
          />
        </div>
        <button
          class="NewReptileForm-submit"
          type="submit"
        >
          Add Reptile
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of edit reptile 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="EditReptileForm"
    >
      <h1>
        Edit your reptile
      </h1>
      <form>
        <div
          class="EditReptileForm-form-group"
        >
          <label
            class="EditReptileForm-label"
            for="name"
          >
            Name:
          </label>
          <input
            class="EditReptileForm-input"
            id="name"
            name="name"
            type="text"
            value="two crows"
          />
        </div>
        <div
          class="EditReptileForm-form-group"
        >
          <label
            class="EditReptileForm-label"
            for="subspecies"
          >
            Subspecies:
          </label>
          <input
            class="EditReptileForm-input"
            id="subspecies"
            name="subspecies"
            type="text"
            value="two of them"
          />
        </div>
        <div
          class="EditReptileForm-form-group"
        >
          <label
            class="EditReptileForm-label"
            for="birthday"
          >
            Birthday:
          </label>
          <input
            class="EditReptileForm-input"
            id="birthday"
            name="birthday"
            type="date"
            value="2015-10-29"
          />
        </div>
        <div
          class="EditReptileForm-form-group"
        >
          <label
            class="EditReptileForm-label"
            for="imgUrl"
          >
            Picture:
          </label>
          <input
            class="EditReptileForm-input"
            id="imgUrl"
            name="imgUrl"
            type="text"
            value=""
          />
        </div>
        <button
          class="EditReptileForm-button submit"
          type="submit"
        >
          Edit
        </button>
        <button
          class="EditReptileForm-button delete"
        >
          Delete reptile
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of edit user page 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="EditUserForm"
    >
      <h1>
        Edit your information
      </h1>
      <form>
        <div
          class="EditUserForm-form-group"
        >
          <label
            class="EditUserForm-label"
            for="username"
          >
            Username:
          </label>
          <input
            class="EditUserForm-input"
            id="username"
            name="username"
            type="text"
            value="rickSanchez"
          />
        </div>
        <div
          class="EditUserForm-form-group"
        >
          <label
            class="EditUserForm-label"
            for="password"
          >
            Password:
          </label>
          <input
            class="EditUserForm-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="EditUserForm-form-group"
        >
          <label
            class="EditUserForm-label"
            for="email"
          >
            Email:
          </label>
          <input
            class="EditUserForm-input"
            id="email"
            name="email"
            type="email"
            value="rickman@qmail.com"
          />
        </div>
        <button
          class="EditUserForm-submit"
          type="submit"
        >
          Edit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of food finder 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Foods"
    >
      <h1>
        Food for your reptiles!
      </h1>
      <p>
        Pick foods you have or can easily get. We recommend having a large variety of foods to ensure a balanced diet.
      </p>
      <div
        class="FoodOptions"
      >
        <p>
          Use these options to filter foods
        </p>
        <div
          class="FoodOptions-group"
        >
          <label
            class="FoodOptions-label"
            for="species"
          >
            Species
          </label>
          <select
            class="FoodOptions-input"
            id="species"
            name="species"
          >
            <option
              value="blue-tongue-skink"
            >
              Blue Tongue Skink
            </option>
          </select>
        </div>
        <div
          class="FoodOptions-group"
        >
          <label
            class="FoodOptions-label"
            for="search"
          >
            Search foods
          </label>
          <input
            class="FoodOptions-input"
            id="search"
            name="search"
            type="text"
            value=""
          />
        </div>
        <div
          class="FoodOptions-group"
        >
          <label
            class="FoodOptions-label"
            for="frequency"
          >
            Food group
          </label>
          <select
            class="FoodOptions-input"
            id="type"
            name="type"
          >
            <option
              value=""
            >
              --
            </option>
            <option
              value="protein"
            >
              Protein
            </option>
            <option
              value="vegetable"
            >
              Vegetables/Greens
            </option>
            <option
              value="fruit"
            >
              Fruit
            </option>
            <option
              value="supplement"
            >
              Supplements
            </option>
          </select>
        </div>
        <div
          class="FoodOptions-group"
        >
          <label
            class="FoodOptions-label"
            for="treats"
          >
            Treats
          </label>
          <input
            class="FoodOptions-checkbox"
            id="treats"
            name="treats"
            type="checkbox"
            value="false"
          />
        </div>
      </div>
      <table
        class="FoodTable"
      >
        <thead
          class="FoodTable-header"
        >
          <tr
            class="FoodTable-header-cells"
          >
            <td>
              Image
            </td>
            <td>
              Name
            </td>
            <td>
              Type
            </td>
            <td>
              How often?
            </td>
            <td>
              Treat?
            </td>
            <td>
              Tips
            </td>
            <td>
              Add / Remove
            </td>
          </tr>
        </thead>
        <tbody
          class="FoodTable-body"
        />
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of login page 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Login"
    >
      <h1>
        Login to your Reptifeed account
      </h1>
      <form>
        <div
          class="Login-form-group"
        >
          <label
            class="Login-label"
            for="username"
          >
            Username:
          </label>
          <input
            class="Login-input"
            id="username"
            name="username"
            type="text"
            value=""
          />
        </div>
        <div
          class="Login-form-group"
        >
          <label
            class="Login-label"
            for="password"
          >
            Password:
          </label>
          <input
            class="Login-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="Login-submit"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of pantry 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Pantry"
    >
      <div>
        <h1>
          My Pantry
        </h1>
        <a
          class="Pantry-link"
          href="/foods"
        >
          Add foods to pantry
        </a>
      </div>
      <div
        class="PantryCard vegetable"
      >
        <div
          class="PantryCard-img-wrapper"
        >
          <img
            alt="arugula"
            src="https://halfyourplate-4kgxi1gvwldjzby.stackpathdns.com/wp-content/uploads/2014/12/arugula_small.gif"
          />
        </div>
        <h2>
          arugula
        </h2>
        <p>
          Feed often
        </p>
        <p>
          Treat?: 
          <span>
            No
          </span>
        </p>
        <p>
          Tips: N/A
        </p>
        <button
          class="PantryCard-removebtn"
        >
          I ran out
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of profile page 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Profile"
    >
      <h1
        class="Profile-header"
      >
        rickSanchez
      </h1>
      <a
        class="Profile-todo-link"
        href="/users/1/todos"
      >
        Click here for next steps!
      </a>
      <div
        class="Profile-section reptile"
      >
        <div
          class="Profile-options"
        >
          <a
            href="/users/1/reptiles"
          >
            See your reptiles
          </a>
          <a
            href="/users/1/reptiles/add"
          >
            Add a reptile
          </a>
        </div>
        <img
          alt=""
          class="Profile-icon"
          src="reptile.png"
        />
      </div>
      <div
        class="Profile-section pantry"
      >
        <img
          alt=""
          class="Profile-icon"
          src="pantry.png"
        />
        <div
          class="Profile-options"
        >
          <a
            href="/users/1/pantry"
          >
            See your pantry
          </a>
          <a
            href="/foods"
          >
            Add foods to pantry
          </a>
        </div>
      </div>
      <div
        class="Profile-section user"
      >
        <div
          class="Profile-options"
        >
          <a
            href="/users/1/edit"
          >
            Edit your profile
          </a>
          <a
            href="/"
          >
            Delete account
          </a>
        </div>
        <img
          alt=""
          class="Profile-icon"
          src="user.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of registration page 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Register"
    >
      <h1>
        Register for your Reptifeed account
      </h1>
      <form>
        <div
          class="Register-form-group"
        >
          <label
            class="Register-label"
            for="username"
          >
            Username:
          </label>
          <input
            class="Register-input"
            id="username"
            name="username"
            type="text"
            value=""
          />
        </div>
        <div
          class="Register-form-group"
        >
          <label
            class="Register-label"
            for="password"
          >
            Password:
          </label>
          <input
            class="Register-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="Register-form-group"
        >
          <label
            class="Register-label"
            for="email"
          >
            Email:
          </label>
          <input
            class="Register-input"
            id="email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <button
          class="Register-submit"
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of reptile page 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Reptile"
    >
      <div
        class="Reptile-card"
      >
        <div
          class="Reptile-card-img-wrapper"
        >
          <img
            alt="two crows"
            src=""
          />
        </div>
        <div>
          <div
            class="Reptile-card-info"
          >
            <h1>
              two crows
            </h1>
            <h4>
              Species: crows
            </h4>
            <h4>
              Subspecies: two of them
            </h4>
            <h5>
              Born on 2015-10-29
            </h5>
          </div>
          <button
            class="edit"
          >
            Edit
          </button>
        </div>
      </div>
      <div
        class="Reptile-button-wrapper"
      >
        <button
          class="feed"
        >
          Feed
        </button>
        <button
          class="treat"
        >
          Give a treat
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of reptiles page 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Reptiles"
    >
      <h1>
        My Reptiles
      </h1>
      <p>
        Click on a reptile to feed them.
      </p>
      <div
        class="ReptileCard"
      >
        <img
          alt="two crows"
          src=""
        />
        <h4>
          two crows
        </h4>
        <h5>
          crows
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot of todos 1`] = `
<DocumentFragment>
  <div
    id="main"
  >
    <div
      class="Todos"
    >
      <h1>
        Todos
      </h1>
      <h2>
        Click on one to get started
      </h2>
      <div
        class="Todos-wrapper"
      >
        <div
          class="Todos-wrapper essential-wrapper"
        >
          <h4>
            Essential
          </h4>
          <a
            class="Todos-todo essential"
            href="/foods"
          >
            Add protein eaten often
          </a>
          <a
            class="Todos-todo essential"
            href="/foods"
          >
            Add a treat eaten moderately
          </a>
          <a
            class="Todos-todo essential"
            href="/foods"
          >
            Add at least two supplements
          </a>
        </div>
        <div
          class="Todos-wrapper niceToHave-wrapper"
        >
          <h4>
            Nice to Have
          </h4>
          <a
            class="Todos-todo niceToHave"
            href="/foods"
          >
            Add a vegetable eaten moderately
          </a>
          <a
            class="Todos-todo niceToHave"
            href="/foods"
          >
            Add a vegetable eaten occasionally
          </a>
          <a
            class="Todos-todo niceToHave"
            href="/foods"
          >
            Add a protein eaten moderately
          </a>
          <a
            class="Todos-todo niceToHave"
            href="/foods"
          >
            Add a protein eaten occasionally
          </a>
          <a
            class="Todos-todo niceToHave"
            href="/foods"
          >
            Add a treat eaten occasionally
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
